applications/GeoMechanicsApplication/tests/cpp_tests/test_solving_strategy_factory.cpp

```diff
@@ -0,0 +1,121 @@
+        "echo_level":                         1,
+        "clear_storage":                      false,
+        "compute_reactions":                  true,
+        "move_mesh_flag":                     false,
+        "reform_dofs_at_each_step":           false,
+        "nodal_smoothing":                    false,
+        "block_builder":                      true,
+        "solution_type":                      "Quasi-Static",
+        "scheme_type":                        "Backward_Euler",
+        "reset_displacements":                false,
+        "newmark_beta":                       0.25,
+        "newmark_gamma":                      0.5,
+        "newmark_theta":                      0.5,
+        "rayleigh_m":                         0.0,
+        "rayleigh_k":                         0.0,
+        "strategy_type":                      "newton_raphson_strategy",
+        "convergence_criterion":              "displacement_criterion",
+        "displacement_relative_tolerance":    1.0E-4,
+        "displacement_absolute_tolerance":    1.0E-9,
+        "residual_relative_tolerance":        1.0E-4,
+        "residual_absolute_tolerance":        1.0E-9,
+        "water_pressure_relative_tolerance":  1.0E-4,
+        "water_pressure_absolute_tolerance":  1.0E-9,
+        "min_iterations":                     6,
+        "max_iterations":                     15,
+        "number_cycles":                      100,
+        "reduction_factor":                   0.5,
+        "increase_factor":                    2.0,
+        "desired_iterations":                 4,
+        "max_radius_factor":                  10.0,
+        "min_radius_factor":                  0.1,
+        "calculate_reactions":                true,
+        "max_line_search_iterations":         5,
+        "first_alpha_value":                  0.5,
+        "second_alpha_value":                 1.0,
+        "min_alpha":                          0.1,
+        "max_alpha":                          2.0,
+        "line_search_tolerance":              0.5,
+        "rotation_dofs":                      true,
+        "linear_solver_settings":             {
+            "solver_type":     "sparse_lu",
+            "scaling":         true
+        },
+        "problem_domain_sub_model_part_list": ["Clay_after_excavation","Excavated_clay"],
+        "processes_sub_model_part_list":      ["Side_sliders","Bottom_fixed","Head_line","Bottom_pressure","Gravitational_load"],
+        "body_domain_sub_model_part_list":    ["Clay_after_excavation","Excavated_clay"]
+    }
+)";
+
+KRATOS_TEST_CASE_IN_SUITE(CreateSolvingStrategy_Throws_WhenNoStrategyTypeIsDefined, KratosGeoMechanicsFastSuite)
+{
+    Model model;
+    model.CreateModelPart("dummy");
+
+    Parameters parameters{};
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(auto test = SolvingStrategyFactoryType::Create(
+            parameters, model.GetModelPart("dummy")), "The parameter strategy_type is undefined, aborting.")
```

Comment:
I would assign the return value and use that when you try to create the solving strategy:
```suggestion
    auto& dummy_model_part = model.CreateModelPart("dummy");

    Parameters parameters{};

    KRATOS_EXPECT_EXCEPTION_IS_THROWN(auto test = SolvingStrategyFactoryType::Create(
            parameters, dummy_model_part), "The parameter strategy_type is undefined, aborting.")
```

Reply:
Ah, I missed that the create function returned a reference to the model part, good suggestion